middleware/capacity.go

```diff
@@ -0,0 +1,20 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+	"github.com/livepeer/catalyst-api/pipeline"
+)
+
+const MAX_JOBS_IN_FLIGHT = 10
+
+func HasCapacity(vodEngine *pipeline.Coordinator, next httprouter.Handle) httprouter.Handle {
+	return func(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+		if vodEngine.Jobs.Size() >= MAX_JOBS_IN_FLIGHT {
```

Comment:
This will count both catalyst and MediaConvert jobs, while we only need to limit on catalyst jobs. WDYT of having a function in the cache that returns a slice of all keys instead and here only count catalyst jobs? Might be worth a helper in the coordinator for that

Reply:
Good point, will update